### PR TITLE
For #27: Added test for Npm Tgz metadata update

### PR DESCRIPTION
**Pull request by paulodamaso**
For #27: Added test for Npm Tgz metadata update
- Created `NpmTest` with test for updating metadata when uploading tgz archive
- Created method placeholder in `Npm`
- Left puzzle for implementation

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#35](https://github.com/artipie/npm-adapter/issues/35) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/697c371d-d9ad-464a-9c47-d4611f34dd1b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by carlosmiranda** [comment]
@g4s8 do I need to do anything here...?

**Comment by g4s8** [comment]
@carlosmiranda sure, you are reviewer here.

File: src/test/java/com/artipie/npm/NpmTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                        .add(NpmTest.DESCRIPTION, "A sample project")
+                        .build()
+                        .toString()
+                    ).bytes()
+                )
+            )
+        );
+        npm.updateMetaFile(key, new TgzArchive(metadata.toString()));
+        MatcherAssert.assertThat(
+            storage.value(key),
```
Comment (carlosmiranda) [inline src/test/java/com/artipie/npm/NpmTest.java]: @paulodamaso this is never going to work, as mentioned by @Sammers21 . We are comparing `CompletableFuture` with `Flowable` here, besides `Flowable` inherits `equals()` from `Object`.

**Comment by paulodamaso** [comment]
@carlosmiranda Fixed the assertion, please take a look

**Comment by carlosmiranda** [comment]
@g4s8 looks good to me

**Comment by carlosmiranda** [comment]
@g4s8 ping

**Comment by carlosmiranda** [comment]
@g4s8 ping

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Pull request merged**